common/component/redis/redis.go

```diff
@@ -132,9 +142,9 @@ func ParseClientFromProperties(properties map[string]string, componentType metad
 		settings.Concurrency = 10
 	}
 
-	err = settings.Decode(properties)
-	if err != nil {
-		return nil, nil, fmt.Errorf("redis client configuration error: %w", err)
+	decodeErr := settings.Decode(properties)
```

Comment:
It's habit for readability, but also because the "shadow" linter - which didn't necessarily trigger here, but that linter taught me to always have dedicated error names.